Project 2

In [34]:
from sklearn.linear_model import LogisticRegressionCV
from sklearn.preprocessing import StandardScaler, LabelEncoder, LabelBinarizer
from sklearn.datasets import load_iris
from sklearn.metrics import confusion_matrix, accuracy_score, classification_report
from sklearn.model_selection import cross_val_score, train_test_split
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

pd.options.display.width = 180

In [35]:
sample = pd.read_csv("sample_sub_reg.csv")
train = pd.read_csv("train.csv")
test = pd.read_csv("test.csv")

In [36]:
def datachecks(df):
    display("HEAD and TAIL: ",df.head(3), df.tail(3))
    display("NULL VALUES: ",df.isnull().sum(),"IS NA: ", df.isna().sum().sum())
    display("DESCRIPTIONS: ", df.describe().T, df.info())
    
datachecks(train)

'HEAD and TAIL: '

Id        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street Alley Lot Shape Land Contour    ...     Screen Porch Pool Area Pool QC Fence Misc Feature Misc Val  \
0  109  533352170           60        RL           NaN     13517   Pave   NaN       IR1          Lvl    ...                0         0     NaN   NaN          NaN        0   
1  544  531379050           60        RL          43.0     11492   Pave   NaN       IR1          Lvl    ...                0         0     NaN   NaN          NaN        0   
2  153  535304180           20        RL          68.0      7922   Pave   NaN       Reg          Lvl    ...                0         0     NaN   NaN          NaN        0   

  Mo Sold Yr Sold  Sale Type  SalePrice  
0       3    2010        WD      130500  
1       4    2009        WD      220000  
2       1    2010        WD      109000  

[3 rows x 81 columns]

Id        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street Alley Lot Shape Land Contour    ...     Screen Porch Pool Area Pool QC Fence Misc Feature Misc Val  \
2048  916  909253010           50        RL          57.0      7558   Pave   NaN       Reg          Bnk    ...                0         0     NaN   NaN          NaN        0   
2049  639  535179160           20        RL          80.0     10400   Pave   NaN       Reg          Lvl    ...                0         0     NaN   NaN          NaN        0   
2050   10  527162130           60        RL          60.0      7500   Pave   NaN       Reg          Lvl    ...                0         0     NaN   NaN          NaN        0   

     Mo Sold Yr Sold  Sale Type  SalePrice  
2048       3    2009        WD      177000  
2049      11    2009        WD      144000  
2050       6    2010        WD      189000  

[3 rows x 81 columns]

'NULL VALUES: '

Id                   0
PID                  0
MS SubClass          0
MS Zoning            0
Lot Frontage       330
Lot Area             0
Street               0
Alley             1911
Lot Shape            0
Land Contour         0
Utilities            0
Lot Config           0
Land Slope           0
Neighborhood         0
Condition 1          0
Condition 2          0
Bldg Type            0
House Style          0
Overall Qual         0
Overall Cond         0
Year Built           0
Year Remod/Add       0
Roof Style           0
Roof Matl            0
Exterior 1st         0
Exterior 2nd         0
Mas Vnr Type        22
Mas Vnr Area        22
Exter Qual           0
Exter Cond           0
                  ... 
Half Bath            0
Bedroom AbvGr        0
Kitchen AbvGr        0
Kitchen Qual         0
TotRms AbvGrd        0
Functional           0
Fireplaces           0
Fireplace Qu      1000
Garage Type        113
Garage Yr Blt      114
Garage Finish      114
Garage Cars          1
Garage Area

'IS NA: '

9822

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2051 entries, 0 to 2050
Data columns (total 81 columns):
Id                 2051 non-null int64
PID                2051 non-null int64
MS SubClass        2051 non-null int64
MS Zoning          2051 non-null object
Lot Frontage       1721 non-null float64
Lot Area           2051 non-null int64
Street             2051 non-null object
Alley              140 non-null object
Lot Shape          2051 non-null object
Land Contour       2051 non-null object
Utilities          2051 non-null object
Lot Config         2051 non-null object
Land Slope         2051 non-null object
Neighborhood       2051 non-null object
Condition 1        2051 non-null object
Condition 2        2051 non-null object
Bldg Type          2051 non-null object
House Style        2051 non-null object
Overall Qual       2051 non-null int64
Overall Cond       2051 non-null int64
Year Built         2051 non-null int64
Year Remod/Add     2051 non-null int64
Roof Style         20

'DESCRIPTIONS: '

count          mean           std          min          25%          50%           75%          max
Id               2051.0  1.474034e+03  8.439808e+02          1.0        753.5       1486.0  2.198000e+03       2930.0
PID              2051.0  7.135900e+08  1.886918e+08  526301100.0  528458140.0  535453200.0  9.071801e+08  924152030.0
MS SubClass      2051.0  5.700878e+01  4.282422e+01         20.0         20.0         50.0  7.000000e+01        190.0
Lot Frontage     1721.0  6.905520e+01  2.326065e+01         21.0         58.0         68.0  8.000000e+01        313.0
Lot Area         2051.0  1.006521e+04  6.742489e+03       1300.0       7500.0       9430.0  1.151350e+04     159000.0
Overall Qual     2051.0  6.112140e+00  1.426271e+00          1.0          5.0          6.0  7.000000e+00         10.0
Overall Cond     2051.0  5.562165e+00  1.104497e+00          1.0          5.0          5.0  6.000000e+00          9.0
Year Built       2051.0  1.971709e+03  3.017789e+01       1872.0       1953.5       1974.0  2.001000e+03       2010.0
Year Remod/Add   2051.0  1.984190e+03  2.103625e+01       1950.0       1964.5       1993.0  2.004000e+03       2010.0
Mas Vnr Area     2029.0  9.969591e+01  1.749631e+02          0.0          0.0          0.0  1.610000e+02       1600.0
BsmtFin SF 1     2050.0  4.423005e+02  4.612041e+02          0.0          0.0        368.0  7.337500e+02       5644.0
BsmtFin SF 2     2050.0  4.795902e+01  1.650009e+02          0.0          0.0          0.0  0.000000e+00       1474.0
Bsmt Unf SF      2050.0  5.677283e+02  4.449548e+02          0.0        220.0        474.5  8.110000e+02       2336.0
Total Bsmt SF    2050.0  1.057988e+03  4.494107e+02          0.0        793.0        994.5  1.318750e+03       6110.0
1st Flr SF       2051.0  1.164488e+03  3.964469e+02        334.0        879.5       1093.0  1.405000e+03       5095.0
2nd Flr SF       2051.0  3.293291e+02  4.256710e+02          0.0          0.0          0.0  6.925000e+02       1862.0
Low Qual Fin SF  2051.0  5.512921e+00  5.106887e+01          0.0          0.0          0.0  0.000000e+00       1064.0
Gr Liv Area      2051.0  1.499330e+03  5.004478e+02        334.0       1129.0       1444.0  1.728500e+03       5642.0
Bsmt Full Bath   2049.0  4.275256e-01  5.226732e-01          0.0          0.0          0.0  1.000000e+00          3.0
Bsmt Half Bath   2049.0  6.344558e-02  2.517052e-01          0.0          0.0          0.0  0.000000e+00          2.0
Full Bath        2051.0  1.577279e+00  5.492794e-01          0.0          1.0          2.0  2.000000e+00          4.0
Half Bath        2051.0  3.710385e-01  5.010427e-01          0.0          0.0          0.0  1.000000e+00          2.0
Bedroom AbvGr    2051.0  2.843491e+00  8.266183e-01          0.0          2.0          3.0  3.000000e+00          8.0
Kitchen AbvGr    2051.0  1.042906e+00  2.097900e-01          0.0          1.0          1.0  1.000000e+00          3.0
TotRms AbvGrd    2051.0  6.435885e+00  1.560225e+00          2.0          5.0          6.0  7.000000e+00         15.0
Fireplaces       2051.0  5.909313e-01  6.385163e-01          0.0          0.0          1.0  1.000000e+00          4.0
Garage Yr Blt    1937.0  1.978708e+03  2.544109e+01       1895.0       1961.0       1980.0  2.002000e+03       2207.0
Garage Cars      2050.0  1.776585e+00  7.645374e-01          0.0          1.0          2.0  2.000000e+00          5.0
Garage Area      2050.0  4.736717e+02  2.159346e+02          0.0        319.0        480.0  5.760000e+02       1418.0
Wood Deck SF     2051.0  9.383374e+01  1.285494e+02          0.0          0.0          0.0  1.680000e+02       1424.0
Open Porch SF    2051.0  4.755680e+01  6.674724e+01          0.0          0.0         27.0  7.000000e+01        547.0
Enclosed Porch   2051.0  2.257192e+01  5.984511e+01          0.0          0.0          0.0  0.000000e+00        432.0
3Ssn Porch       2051.0  2.591419e+00  2.522961e+01          0.0          0.0          0.0  0.000000e+00        508.0
Screen

None

In [4]:
null_columns = train.columns[train.isnull().any()]

In [5]:
train[null_columns].isnull().sum()

Lot Frontage       330
Alley             1911
Mas Vnr Type        22
Mas Vnr Area        22
Bsmt Qual           55
Bsmt Cond           55
Bsmt Exposure       58
BsmtFin Type 1      55
BsmtFin SF 1         1
BsmtFin Type 2      56
BsmtFin SF 2         1
Bsmt Unf SF          1
Total Bsmt SF        1
Bsmt Full Bath       2
Bsmt Half Bath       2
Fireplace Qu      1000
Garage Type        113
Garage Yr Blt      114
Garage Finish      114
Garage Cars          1
Garage Area          1
Garage Qual        114
Garage Cond        114
Pool QC           2042
Fence             1651
Misc Feature      1986
dtype: int64

Lot Frontage, Alley, Mas Vnr Area        22
Bsmt Qual           55
Bsmt Cond           55
Bsmt Exposure       58
BsmtFin Type 1      55
BsmtFin SF 1         1
BsmtFin Type 2      56
BsmtFin SF 2         1
Bsmt Unf SF          1
Total Bsmt SF        1
Bsmt Full Bath       2
Bsmt Half Bath       2
Fireplace Qu      1000 - Fireplace quality
Garage Type        113
Garage Yr Blt      114
Garage Finish      114
Garage Cars          1
Garage Area          1
Garage Qual        114
Garage Cond        114
Pool QC           2042
Fence             1651
Misc Feature      1986

In [6]:
obj_cols = [x for x in train.columns if train[x].dtype == object]

In [8]:
# def dummied(train, test):
#     l = ['MS SubClass']
#     for i in train.columns:
#         if train[i].dtype == object:
#             l.append[i]
    
    
        
# going to try LabelEncoder instead        

In [9]:
train.Alley.unique()

array([nan, 'Pave', 'Grvl'], dtype=object)

In [10]:
# drop nan in alley and binarize?

In [11]:
for x in train[obj_cols]:
    print(x,"\n", train[x].unique())

MS Zoning 
 ['RL' 'RM' 'FV' 'C (all)' 'A (agr)' 'RH' 'I (all)']
Street 
 ['Pave' 'Grvl']
Alley 
 [nan 'Pave' 'Grvl']
Lot Shape 
 ['IR1' 'Reg' 'IR2' 'IR3']
Land Contour 
 ['Lvl' 'HLS' 'Bnk' 'Low']
Utilities 
 ['AllPub' 'NoSeWa' 'NoSewr']
Lot Config 
 ['CulDSac' 'Inside' 'Corner' 'FR2' 'FR3']
Land Slope 
 ['Gtl' 'Sev' 'Mod']
Neighborhood 
 ['Sawyer' 'SawyerW' 'NAmes' 'Timber' 'Edwards' 'OldTown' 'BrDale'
 'CollgCr' 'Somerst' 'Mitchel' 'StoneBr' 'NridgHt' 'Gilbert' 'Crawfor'
 'IDOTRR' 'NWAmes' 'Veenker' 'MeadowV' 'SWISU' 'NoRidge' 'ClearCr'
 'Blmngtn' 'BrkSide' 'NPkVill' 'Blueste' 'GrnHill' 'Greens' 'Landmrk']
Condition 1 
 ['RRAe' 'Norm' 'PosA' 'Artery' 'Feedr' 'PosN' 'RRAn' 'RRNe' 'RRNn']
Condition 2 
 ['Norm' 'RRNn' 'Feedr' 'Artery' 'PosA' 'PosN' 'RRAe' 'RRAn']
Bldg Type 
 ['1Fam' 'TwnhsE' 'Twnhs' '2fmCon' 'Duplex']
House Style 
 ['2Story' '1Story' '1.5Fin' 'SFoyer' 'SLvl' '2.5Unf' '2.5Fin' '1.5Unf']
Roof Style 
 ['Gable' 'Hip' 'Flat' 'Mansard' 'Shed' 'Gambrel']
Roof Matl 
 ['CompShg' 

In [16]:
df = train.drop(columns="SalePrice")
target = train['SalePrice']

In [31]:
le = LabelEncoder

0        NaN
1        NaN
2        NaN
3        NaN
4        NaN
5        NaN
6        NaN
7        NaN
8        NaN
9        NaN
10       NaN
11       NaN
12       NaN
13      Pave
14       NaN
15       NaN
16      Grvl
17       NaN
18       NaN
19       NaN
20       NaN
21       NaN
22       NaN
23       NaN
24       NaN
25       NaN
26       NaN
27      Grvl
28       NaN
29       NaN
        ... 
2021     NaN
2022     NaN
2023     NaN
2024     NaN
2025     NaN
2026     NaN
2027     NaN
2028     NaN
2029     NaN
2030    Pave
2031     NaN
2032     NaN
2033     NaN
2034     NaN
2035     NaN
2036     NaN
2037     NaN
2038     NaN
2039     NaN
2040     NaN
2041     NaN
2042    Grvl
2043     NaN
2044     NaN
2045     NaN
2046     NaN
2047     NaN
2048     NaN
2049     NaN
2050     NaN
Name: Alley, Length: 2051, dtype: object

In [37]:
train[obj_cols].columns

Index(['MS Zoning', 'Street', 'Alley', 'Lot Shape', 'Land Contour', 'Utilities', 'Lot Config', 'Land Slope', 'Neighborhood', 'Condition 1', 'Condition 2', 'Bldg Type',
       'House Style', 'Roof Style', 'Roof Matl', 'Exterior 1st', 'Exterior 2nd', 'Mas Vnr Type', 'Exter Qual', 'Exter Cond', 'Foundation', 'Bsmt Qual', 'Bsmt Cond',
       'Bsmt Exposure', 'BsmtFin Type 1', 'BsmtFin Type 2', 'Heating', 'Heating QC', 'Central Air', 'Electrical', 'Kitchen Qual', 'Functional', 'Fireplace Qu', 'Garage Type',
       'Garage Finish', 'Garage Qual', 'Garage Cond', 'Paved Drive', 'Pool QC', 'Fence', 'Misc Feature', 'Sale Type'],
      dtype='object')

In [39]:
df.head()

Id        PID  MS SubClass MS Zoning  Lot Frontage  Lot Area Street Alley Lot Shape Land Contour    ...     3Ssn Porch Screen Porch Pool Area Pool QC Fence Misc Feature  \
0  109  533352170           60        RL           NaN     13517   Pave   NaN       IR1          Lvl    ...              0            0         0     NaN   NaN          NaN   
1  544  531379050           60        RL          43.0     11492   Pave   NaN       IR1          Lvl    ...              0            0         0     NaN   NaN          NaN   
2  153  535304180           20        RL          68.0      7922   Pave   NaN       Reg          Lvl    ...              0            0         0     NaN   NaN          NaN   
3  318  916386060           60        RL          73.0      9802   Pave   NaN       Reg          Lvl    ...              0            0         0     NaN   NaN          NaN   
4  255  906425045           50        RL          82.0     14235   Pave   NaN       IR1          Lvl    ...              0            0         0     NaN   NaN          NaN   

  Misc Val Mo Sold  Yr Sold  Sale Type  
0        0       3     2010        WD   
1        0       4     2009        WD   
2        0       1     2010        WD   
3        0       4     2010        WD   
4        0       3     2010        WD   

[5 rows x 80 columns]

In [ ]:
y_train = target
X_train = df

for 